### PR TITLE
split QualityNomination creation logic by type

### PR DESCRIPTION
**Pull request by Avadhutgiri**
# Description

Refactored the `QualityNomination` controller to improve code clarity and maintainability.  
- **Main change:** Split the large `createNomination` function into smaller, type-specific functions for each nomination type (`suggestion`, `promotion`, `demotion`, `dismissal`, `quality_tag`).
- Each type now has its own validation and processing logic, making the code easier to understand and maintain.
- No changes to the user interface.

**Fixes:** #8297   

# Comments

- All existing tests were executed and passed, confirming that no regressions were introduced.
- Reviewers may want to pay special attention to the new helper functions to ensure all edge cases are still covered.
- No UI changes were made, so no screenshots or videos are included.

# Checklist:

- [x] The code follows the [coding guidelines](https://github.com/omegaup/omegaup/blob/main/frontend/www/docs/Coding-guidelines.md) of omegaUp.
- [x] The tests were executed and all of them passed.
- [ ] If you are creating a feature, the new tests were added. *(Not a new feature, so not applicable)*
- [x] If the change is large (> 200 lines), this PR was split into various Pull Requests. *(This PR only affects backend logic and tests)*

### CODE REVIEW

**Comment by Avadhutgiri** [comment]
Hey @pabloaguilar Can You check if this PR satisfies the solution??

File: frontend/server/src/Controllers/QualityNomination.php [hunk 1]
```diff
@@ -230,12 +230,83 @@ public static function createNomination(
         string $nominationType,
         array $contents
     ): \OmegaUp\DAO\VO\QualityNominations {
+        // Validate user permissions based on nomination type
```
Comment (pabo99) [inline frontend/server/src/Controllers/QualityNomination.php:233]: For each type of nomination, certain required parameters must exist. What do you think about deconstructing the `contents` array right from the start and only sending the necessary parameters for each validation?

File: frontend/server/src/Controllers/QualityNomination.php [hunk 2]
```diff
@@ -230,12 +230,83 @@ public static function createNomination(
...
+    }
+
+    /**
+     * Validates user permissions based on nomination type and contents
+     */
+    public static function validateUserPermissions(
+        \OmegaUp\DAO\VO\Problems $problem,
+        \OmegaUp\DAO\VO\Identities $identity,
+        string $nominationType,
+        array $contents
```
Comment (pabo99) [inline frontend/server/src/Controllers/QualityNomination.php:303]: Based on the previous comment, here we would only need to receive `$beforeAC`, which would be of boolean type. That way, we can avoid parsing in line 308 and also skip line 307, since it will always exist.

The same approach should be applied to the rest of the validation functions.

File: frontend/server/src/Controllers/QualityNomination.php [hunk 3]
```diff
@@ -230,12 +230,83 @@ public static function createNomination(
...
+    ): void {
```
Comment (pabo99) [inline frontend/server/src/Controllers/QualityNomination.php:304]: Here we could avoid a nested conditional by using the guard clause pattern, which means we can invert the condition to look like this:
```php
if ($nominationType === 'demotion'|| $nominationType === 'quality_tag') {
    return;
}
... // Rest of the function
```

File: frontend/server/src/Controllers/QualityNomination.php [hunk 4]
```diff
@@ -230,12 +230,83 @@ public static function createNomination(
...
+            $identity,
+            $nominationType,
+            $contents
+        );
+
+        $qualityNomination = null;
+
+        // Validate and process nomination based on type
+        switch ($nominationType) {
+            case 'suggestion':
```
Comment (pabo99) [inline frontend/server/src/Controllers/QualityNomination.php:245]: Since you're already making this change, you could also add all these nomination types into an Enum. Fortunately, PHP version 8.1 now supports this feature: https://php.watch/versions/8.1/enums#enum-syntax

**Review by pabo99 — COMMENTED** [review]
I left a few suggestions to make the code more readable.
